backend-rust/src/graphql_api.rs

```diff
@@ -839,6 +1028,24 @@ impl ScalarType for UnsignedLong {
     fn to_value(&self) -> Value { Value::Number(self.0.into()) }
 }
 
+#[derive(Debug, thiserror::Error, Clone)]
+enum UnsignedLongParseError {
+    #[error("Negative number cannot be converted to UnsignedLong.")]
+    NotNegative,
+}
+
+impl TryFrom<i64> for UnsignedLong {
+    type Error = UnsignedLongParseError;
+
+    fn try_from(number: i64) -> Result<Self, Self::Error> {
+        if number < 0 {
+            Err(UnsignedLongParseError::NotNegative)?
+        } else {
+            Ok(UnsignedLong(number as u64))
+        }
+    }
+}
```

Comment:
It would maybe make more sense to just define this in terms of `TryFrom` with i64 to u64. Something like this:

```suggestion
impl TryFrom<i64> for UnsignedLong {
    type Error = <u64 as TryFrom<i64>>::Error;

    fn try_from(number: i64) -> Result<Self, Self::Error> {
        Ok(UnsignedLong(number.try_into()?))
    }
}
```

Then you can also remove the error type declared above.